lib/src/iterable_extensions.dart

```diff
@@ -352,6 +352,11 @@ extension IterableExtension<T> on Iterable<T> {
     }
     return null;
   }
+  
+  /// Returns an element at the given [index]
```

Comment:
Going to nitpick here.

The first paragraph of the dartdoc should preferably fit on a single line.

I'd drop the `Returns` (I personally find that it's never the best word to start a DartDoc, and, yes, I know `elementAt` does start with `Returns`).

Maybe
```dart
/// The [index]th element, or `null` if there is no such element.
///
/// Returns the element at position [index] of this iterable, 
/// just like [elementAt], if the iterable has such an element.
/// If this iterable does not have enough elements to have
/// one with the given [index], the `null` value is returned instead,
/// unlike [elementAt] which throws instead.
///
/// The [index] must not be negative.
```